### PR TITLE
SCRUM-43: docker

### PR DESCRIPTION
**Pull request by YP0n**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    volumes:
+      - pg_marketplace:/var/lib/postgresql/data
+  app:
+    build: .
+    ports:
+      - "8080:8080"
+    depends_on:
+      - postgres
+    env_file:
+      - .env
```
Comment (filkovsp) [inline docker-compose.yml:23]: can we have a readme file for this one please?

Comment (filkovsp) [reply docker-compose.yml:23]: we can have a docker file and compose file in `docker` folder 
and readme file would have these instructions: 
- build image with Dockerfile:
    ```shell
    docker build --tag food-backend --file ./docker/Dockerfile .
    ```
- compose and start back-end and db container:
    ```shell
    docker-compose --file ./docker/docker-compose.yml up -d
    ```

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+version: '3.6'
+
+volumes:
+  pg_marketplace:
+
+services:
+  postgres:
+    image: postgres:latest
+    restart: always
+    env_file:
```
Comment (Ne4upara) [inline docker/docker-compose.yml:10]: У меня не прошел такой код. А єтот проходит env_file:
      -.env

Comment (YP0n) [reply docker/docker-compose.yml:10]: Спробуй після змін.

File: src/main/java/com/khutircraftubackend/config/SecurityConfig.java [hunk 3]
```diff
@@ -25,7 +25,7 @@
 public class SecurityConfig {
 
     private final JwtAuthenticationFilter jwtAuthenticationFilter;
-    private final UserDetailsService userDetailsService;
+    private final UserDetailsServicesImpl userDetailsService;
```
Comment (filkovsp) [inline src/main/java/com/khutircraftubackend/config/SecurityConfig.java:28]: why can't we use interface here?
why do we change this config class as part of "docker related story"?

File: docker/docker-compose.yml (hunk 2 above)
Comment (filkovsp) [reply docker/docker-compose.yml:10]: looks good now

File: docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,19 @@
+FROM maven:3.8.5-openjdk-17 AS build
+
+WORKDIR /app
+
+COPY ../pom.xml ./
```
Comment (filkovsp) [inline docker/Dockerfile:5]: shall be just copy everything? 
```sh
COPY . .
```

File: docker/README.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+      docker-compose --file ./docker/docker-compose.yml up -d
+      ```
+
+    - Це створить та запустить всі необхідні контейнери для вашого проекту.
+
+4. **Перевірка роботи**:
+    - Перевірте, чи всі контейнери працюють коректно:
+
+      ```bash
+      docker-compose ps(ls)
```
Comment (filkovsp) [inline docker/README.md:105]: what is the expected output?

File: docker/README.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+This command will:
+
+	•	Build the image (if not already built).
+	•	Start the PostgreSQL database container.
+	•	Start the Spring Boot application container.
+
+### 2. Access the Application
+
+Once the containers are up and running, you can access the application at:
+http://localhost:8080/...
```
Comment (filkovsp) [inline docker/README.md:35]: where does the 8080 come from? 
what if one wants to change it?

File: docker/README.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+    - Це створить та запустить всі необхідні контейнери для вашого проекту.
+
+4. **Перевірка роботи**:
+    - Перевірте, чи всі контейнери працюють коректно:
+
+      ```bash
+      docker-compose ps(ls)
+      ```
+
```
Comment (filkovsp) [inline docker/README.md:107]: what about Dockerfile?
any instructions about that one?

File: docker/README.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+```
+
+This command will stop and remove the containers, but the data stored in the PostgreSQL volume will be preserved.
+
+
+
+
+# Налаштування проекту
+
+## Умови
```
Comment (filkovsp) [inline docker/README.md:52]: this duplicates "## Prerequisites" section

File: docker/README.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+      ```bash
+      git clone <URL_вашого_репозиторію>
+      ```
+
+2. **Перехід до каталогу проекту**:
+    - Перейдіть у каталог проекту, наприклад:
+
+      ```bash
+      cd <назва_каталогу>
+      ```
```
Comment (filkovsp) [inline docker/README.md:90]: too much details.
people who wants/needs to run it, should be aware of these things. imho.

File: src/main/java/com/khutircraftubackend/config/SecurityConfig.java (hunk 3 above)
Comment (YP0n) [reply src/main/java/com/khutircraftubackend/config/SecurityConfig.java:28]: its UserDetailsServicesImpl userDetailsService; that's exactly what we use.
Config...

File: docker/README.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+
+3. **Запуск Docker контейнерів**:
```
Comment (filkovsp) [inline docker/README.md:92]: duplication of "### 3. Stopping the Application" section

File: docker/Dockerfile (hunk 4 above)
Comment (YP0n) [reply docker/Dockerfile:5]: ./

File: docker/README.md (hunk 5 above)
Comment (YP0n) [reply docker/README.md:105]: ps-container, ls-image

File: docker/README.md (hunk 6 above)
Comment (YP0n) [reply docker/README.md:35]: For change welcome in setting.

File: docker/README.md (hunk 9 above)
Comment (YP0n) [reply docker/README.md:90]: who knows
PS this for Ira. I knew you would have questions here.)

**Review by filkovsp — APPROVED** [review]

**Pull request merged**